src/modules/commander/Commander.cpp

```diff
@@ -3952,19 +3961,27 @@ void Commander::data_link_check(bool &status_changed)
 	}
 
 	// AVOIDANCE SYSTEM state check (only if it is enabled)
-	if (_obs_avoid.get() && !_onboard_controller_lost) {
+	if (status_flags.avoidance_system_required && !_onboard_controller_lost) {
 
 		//if avoidance never started
-		if (_datalink_last_heartbeat_avoidance_system == 0 && hrt_elapsed_time(&_avoidance_system_not_started) > 5_s) {
+		if (_datalink_last_heartbeat_avoidance_system == 0
+		    && hrt_elapsed_time(&_avoidance_system_not_started) > _onboard_boot_timeout.get() * 1_s
+		    && avoidance_waiting_count < AVOIDANCE_MAX_TRIALS) {
 			_avoidance_system_not_started = hrt_absolute_time();
 			mavlink_log_info(&mavlink_log_pub, "Waiting for avoidance system to start");
+			avoidance_waiting_count++;
+
+		} else if (avoidance_waiting_count == AVOIDANCE_MAX_TRIALS) {
+			mavlink_log_critical(&mavlink_log_pub, "Avoidance not responding. Try reboot vehicle.");
```

Comment:
So avoidance is a thing that responds? What about `Could not detect avoidance system. Try to reboot the vehicle.`?

Reply:
Yes, following up my previous comment: this is a suggestion to help the user unblocking the situation. Actually instead of `reboot vehicle` should be `power cycle vehicle` to even restart the onboard computer.